### PR TITLE
Issue 13: Added support for connect sequence timeout 

### LINKED ISSUE #13: Add timeout option for connect/handshake

### PR DESCRIPTION
**Pull request by JCash**

### CODE REVIEW

File: websocket/src/handshake.cpp [hunk 1]
```diff
@@ -105,13 +105,13 @@ Result ReceiveHeaders(WebsocketConnection* conn)
     dmSocket::SelectorZero(&selector);
     dmSocket::SelectorSet(&selector, dmSocket::SELECTOR_KIND_READ, conn->m_Socket);
 
-    dmSocket::Result sr = dmSocket::Select(&selector, 200*1000);
+    dmSocket::Result sr = dmSocket::Select(&selector, SOCKET_WAIT_TIMEOUT);
```
Comment (JCash) [inline websocket/src/handshake.cpp:108]: Changed blocking from 200ms to 4ms means a lot higher frame rate for the user.

File: websocket/src/handshake.cpp [hunk 2]
```diff
@@ -105,13 +105,13 @@ Result ReceiveHeaders(WebsocketConnection* conn)
...
 
     if (dmSocket::RESULT_OK != sr)
     {
         if (dmSocket::RESULT_WOULDBLOCK)
         {
-            dmLogWarning("Waiting for socket to be available for reading");
+            DebugLog(1, "Waiting for socket to be available for reading");
```
Comment (JCash) [inline websocket/src/handshake.cpp:114]: No need to spam this at all for the default user

File: websocket/src/script_util.cpp [hunk 3]
```diff
@@ -1,8 +1,8 @@
 #include "script_util.h"
 
-namespace dmWebsocket {
+namespace dmScript {
```
Comment (JCash) [inline websocket/src/script_util.cpp:3]: Seemed more appropriate imho

File: websocket/src/websocket.cpp [hunk 4]
```diff
@@ -534,7 +552,10 @@ static dmExtension::Result OnUpdate(dmExtension::Params* params)
                 conn->m_BufferSize = 0;
             }
 
-            HandleCallback(conn, EVENT_DISCONNECTED, 0, conn->m_BufferSize);
+            if (conn->m_WasConnected)
+            {
+                HandleCallback(conn, EVENT_DISCONNECTED, 0, conn->m_BufferSize);
```
Comment (JCash) [inline websocket/src/websocket.cpp:557]: DOn't call "DISCONNECTED", if it wasn't actually connected before

**Review by britzl — APPROVED** [review]
Great!

**Pull request merged**